storage/flux/table.gen.go.tmpl

```diff
@@ -117,47 +121,128 @@ func new{{.Name}}WindowTable(
 			cur:   cur,
 		},
 		windowEvery: every,
+		createEmpty: createEmpty,
+	}
+	if t.createEmpty {
+		start := int64(bounds.Start)
+		t.nextTS = start + (every - start % every)
 	}
 	t.readTags(tags)
 	t.advance()
 
 	return t
 }
 
-func (t *{{.name}}WindowTable) advance() bool {
+func (t *{{.name}}WindowTable) Do(f func(flux.ColReader) error) error {
+	return t.do(f, t.advance)
+}
+
+// createNextWindow will read the timestamps from the array
+// cursor and construct the values for the next window.
+func (t *{{.name}}WindowTable) createNextWindow() (start, stop *array.Int64, ok bool) {
```

Comment:
Normally I shy away from functions that are only ever called once, but this makes this process much more readable.